src/hooks/custom-hooks/useExtendedOrderDetails.ts

```diff
@@ -95,61 +95,71 @@ const useExtendedOrderDetails = ({
                     : localize('Your payment details'),
                 resultString: this.isBuyOrder
                     ? localize("You've received {{currency}} {{displayAmount}}", {
-                          currency: this.account_currency,
-                          displayAmount: this.amount_display,
+                          currency: this.p2p_order_info?.account_currency,
+                          displayAmount: this.p2p_order_info?.amount_display,
                       })
                     : localize('You sold {{currency}} {{displayAmount}}', {
-                          currency: this.account_currency,
-                          displayAmount: this.amount_display,
+                          currency: this.p2p_order_info?.account_currency,
+                          displayAmount: this.p2p_order_info?.amount_display,
                       }),
                 rightSendOrReceive: this.isBuyOrder ? localize('Receive') : localize('Send'),
             };
         },
         get displayPaymentAmount() {
-            return removeTrailingZeros(
-                FormatUtils.formatMoney(
-                    Number(this.amount_display) * Number(roundOffDecimal(this.rate, setDecimalPlaces(this.rate, 6))),
-                    { currency: this.local_currency as TCurrency }
-                )
-            );
+            if (this.p2p_order_info?.rate) {
+                return removeTrailingZeros(
+                    FormatUtils.formatMoney(
+                        Number(this.p2p_order_info?.amount_display) *
```

Comment:
this.p2p_order_info?.amount_display ?? 0
the formatMoney function might break if its undefined